In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statistics

In [2]:
data=pd.read_csv('D:\Sibina\ICT academy\Case Studies\Case Study#02\StudentsPerformance.csv')

In [6]:
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

# Q1.Find out how many males and females participated in the test

In [7]:
data1=data.groupby("gender").describe()

In [8]:
data1

math score                                                       \
            count       mean        std   min   25%   50%   75%    max   
gender                                                                   
female      518.0  63.376448  16.029928   0.0  54.0  65.0  74.0  100.0   
male        482.0  68.821577  14.556411  27.0  59.0  69.0  79.0  100.0   

       reading score             ...              writing score             \
               count       mean  ...   75%    max         count       mean   
gender                           ...                                         
female         518.0  72.590734  ...  83.0  100.0         518.0  72.467181   
male           482.0  65.545643  ...  76.0  100.0         482.0  63.446058   

                                                  
              std   min   25%   50%   75%    max  
gender                                            
female  14.844842  10.0  64.0  74.0  82.0  100.0  
male    14.227225  15.0  53.0  64.0  74.0  100.0  

[2 rows x 24 columns]

Number of Females is 518 and Number of Males 482


# Q2.What do you think about the students' parental level of education?

In [9]:
data2=data.groupby("parental level of education").describe()

In [10]:
data2

math score                                     \
                                 count       mean        std   min    25%   
parental level of education                                                 
associate's degree               222.0  67.977477  15.108190  26.0  57.00   
bachelor's degree                118.0  69.288136  15.164117  29.0  61.00   
high school                      196.0  61.821429  15.160466   8.0  53.75   
master's degree                   59.0  70.254237  14.843544  40.0  57.00   
some college                     226.0  67.128319  14.871790  11.0  59.00   
some high school                 179.0  63.134078  16.602894   0.0  53.00   

                                               reading score             ...  \
                              50%   75%    max         count       mean  ...   
parental level of education                                              ...   
associate's degree           67.0  80.0  100.0         222.0  71.018018  ...   
bachelor's degree            68.0  79.0  100.0         118.0  73.000000  ...   
high school                  63.0  72.0   99.0         196.0  64.602041  ...   
master's degree              73.0  81.0   95.0          59.0  75.949153  ...   
some college                 68.0  76.0  100.0         226.0  69.566372  ...   
some high school             65.0  74.0   97.0         179.0  66.759777  ...   

                                          writing score                        \
                               75%    max         count       mean        std   
parental level of education                                                     
associate's degree           81.00  100.0         222.0  70.031532  14.260787   
bachelor's degree            82.75  100.0         118.0  73.381356  14.728262   
high school                  74.25   99.0         196.0  62.403061  14.165172   
master's degree              84.50  100.0          59.0  75.677966  13.730711   
some college                 80.00  100.0         226.0  69.035398  15.089911   
some high school             79.00  100.0         179.0  64.888268  15.736197   

                                                             
                              min    25%   50%   75%    max  
parental level of education                                  
associate's degree           35.0  58.25  71.5  80.0  100.0  
bachelor's degree            38.0  62.50  74.0  83.0  100.0  
high school                  15.0  52.00  64.0  73.0  100.0  
master's degree              46.0  67.00  75.0  85.0  100.0  
some college                 19.0  60.00  70.0  79.0   99.0  
some high school             10.0  54.00  66.0  77.0  100.0  

[6 rows x 24 columns]

Majority of Parents are with some college eduaction(who started college but didn’t finish ) and least are with master's degree

# Q3.Who scores the most on average for math, reading and writing based on

# Q3.1.Gender

In [11]:
data1

math score                                                       \
            count       mean        std   min   25%   50%   75%    max   
gender                                                                   
female      518.0  63.376448  16.029928   0.0  54.0  65.0  74.0  100.0   
male        482.0  68.821577  14.556411  27.0  59.0  69.0  79.0  100.0   

       reading score             ...              writing score             \
               count       mean  ...   75%    max         count       mean   
gender                           ...                                         
female         518.0  72.590734  ...  83.0  100.0         518.0  72.467181   
male           482.0  65.545643  ...  76.0  100.0         482.0  63.446058   

                                                  
              std   min   25%   50%   75%    max  
gender                                            
female  14.844842  10.0  64.0  74.0  82.0  100.0  
male    14.227225  15.0  53.0  64.0  74.0  100.0  

[2 rows x 24 columns]

*Highsest average for math,reading & writing are as follows
*math=Male=68.82
*reading score=Female=72.59
*writing score=Female=72.46
So altogether, Females scored more


# Q3.2.Test preparation course

In [12]:
data3=data.groupby("test preparation course").describe()

In [13]:
data3

math score                                         \
                             count      mean        std   min   25%   50%   
test preparation course                                                     
completed                    358.0  69.96648  14.521847  23.0  61.0  70.0   
none                         642.0  63.78972  15.705689   0.0  54.0  64.0   

                                     reading score             ...        \
                          75%    max         count       mean  ...   75%   
test preparation course                                        ...         
completed                79.0  100.0         358.0  74.175978  ...  84.0   
none                     74.0  100.0         642.0  66.417445  ...  76.0   

                               writing score                              \
                           max         count       mean        std   min   
test preparation course                                                    
completed                100.0         358.0  74.684358  13.236412  36.0   
none                     100.0         642.0  64.457944  15.041667  10.0   

                                                   
                          25%   50%    75%    max  
test preparation course                            
completed                67.0  76.0  83.75  100.0  
none                     54.0  65.0  74.00  100.0  

[2 rows x 24 columns]

*Highsest average for math,reading & writing are as follows
*math=Completed=69.96
*reading score=completed=74.17
*writing score=completed=74.68
 those who completed the test preparation course scored more

# Q4.What do you think about the scoring variation for math, reading and writing based on

# Q4.1.Gender

*For math, Males scored more and for Reading & Writing Males Scored More

# Q4.2. Test Preparation course

*for all three sections math,reading and writing those who completed the test preparation course scored more than those didn't completed the course

# Q5.The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.

*There are 1000 students, so top 25% students are first 250 students who scored top in maths

In [18]:
data.sort_values(by="math score",ascending=False)

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
2    female        group B            some high school  free/reduced   
51   female        group C            some high school  free/reduced   
3    female        group B                some college      standard   
1    female        group B                 high school  free/reduced   
0    female        group C            some high school  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
2                      none          18             32             28  
51                     none          13             41             51  
3                      none          11             38             32  
1                      none           8             24             23  
0                      none           0             17             10  

[1000 rows x 8 columns]

*To Display First Students only

In [19]:
data.head(250)

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
245  female        group B                 high school  free/reduced   
246  female        group C          associate's degree  free/reduced   
247  female        group C            some high school  free/reduced   
248  female        group C                some college  free/reduced   
249  female        group C          associate's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
245                    none          64             73             71  
246                    none          64             73             68  
247               completed          64             79             77  
248               completed          64             85             85  
249                    none          64             64             70  

[250 rows x 8 columns]